src/azure-cli/azure/cli/command_modules/resource/_params.py

```diff
@@ -565,13 +565,13 @@ def load_arguments(self, _):
             c.argument('notes', help='Notes about this lock.')
 
     with self.argument_context('managedapp') as c:
-        c.argument('resource_group_name', arg_type=resource_group_name_type, help='the resource group of the managed application', id_part='resource_group')
-        c.argument('application_name', options_list=['--name', '-n'], id_part='name')
+        c.argument('resource_group_name', arg_type=resource_group_name_type, required=True, help='the resource group of the managed application', id_part='resource_group')
+        c.argument('application_name', options_list=['--name', '-n'], required=True, id_part='name')
```

Comment:
In fact, `--resource-group-name` and `--application-name` parameters should not be required in all cases. For example, if users passe in `--ids`, he does not need to pass in `--resource-group-name` and `--application-name` anymore. 
It is suggested that in this case, we can first tell customers `--resource-group-name` and `--application-name` need to be used together in the help message

Reply:
Received, I will complete the changes in the next cycle